[Jeriel Chan Zhi Yang] iP

Nimastic opened the pull request:
# JerielPro - A Personal Task Manager

> "The human brain has a built-in limit on the number of discrete thoughts it can entertain at one time. The limit for most individuals is four." – Researchers at Oregon University

JerielPro frees your mind from remembering all the things you need to do. It's:

- Simple to use
- Fast and efficient
- Helps manage your tasks

Here’s how you can get started:

1. [Download JerielPro](https://example.com/download).
2. Add tasks you want to track.
3. Let JerielPro manage them for you.

And the best part? It's **FREE**! 🎉

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you're a Java programmer, here's the `main` method to get you started:

```java
public class Main {
    public static void main(String[] args) {
        System.out.println("Hello, JerielPro!");
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Nimastic committed:
First Commit

Nimastic committed:
Level-0

Nimastic committed:
Level-1

Nimastic committed:
Level-2

Nimastic committed:
Level-3

Nimastic committed:
Level-4

Nimastic committed:
Level-5

Nimastic committed:
Level-6

Nimastic committed:
Level-6

Nimastic committed:
Add task saving and loading functionality

Nimastic committed:
Merge pull request #1 from Nimastic/branch-level-7

Add task saving and loading functionality

Nimastic committed:
Add support for saving tasks with dates and times

Nimastic committed:
Merge pull request #2 from Nimastic/branch-level-8

Add support for saving tasks with dates and times

Nimastic committed:
Implement Command classes for various operations

Nimastic committed:
Merge branch 'master' into branch-A-MoreOOP

Nimastic committed:
Merge pull request #4 from Nimastic/A-MoreOOP

Add More OOP

Nimastic committed:
Refactor project structure into packages

Nimastic committed:
Merge pull request #5 from Nimastic/branch-A-Packages

Refactor project structure into packages

Nimastic committed:
Merge branch 'add-gradle-support'

Nimastic committed:
Add Gradle support for the Jeriel project

Nimastic committed:
Add JUnit tests for Todo and Event classes

Nimastic committed:
Merge pull request #6 from Nimastic/branch-A-JUnit

Add JUnit tests for Todo and Event classes

Nimastic committed:
Add support for creating an executable JAR file

Nimastic committed:
Merge pull request #7 from Nimastic/branch-A-Jar

Add support for creating an executable JAR file

Nimastic committed:
Add JavaDoc comments to codebase

Nimastic committed:
Merge pull request #8 from Nimastic/branch-A-JavaDoc

Add JavaDoc comments to codebase

Nimastic committed:
Apply coding standard improvements across codebase

Naming Conventions:
Classes and Interfaces: Use PascalCase (e.g., MyClass, TaskList).
Methods and Variables: Use camelCase (e.g., calculateTotal, taskCount).
Constants: Use UPPER_SNAKE_CASE (e.g., MAX_SIZE, DEFAULT_VALUE).

Nimastic committed:
Merge pull request #9 from Nimastic/branch-A-CodingStandard

Apply coding standard improvements across codebase

Nimastic committed:
Add find feature

Nimastic committed:
Merge pull request #10 from Nimastic/branch-Level-9

Add find feature

Nimastic committed:
Add CheckStyle and fix coding style violations

- Integrated CheckStyle with Gradle for automated style checks
- Added 'checkstyle.xml' configuration to enforce coding standards
- Fixed coding style violations detected by CheckStyle
- Addressed warnings for Gradle 9.0 compatibility

Tested and ensured build is successful with updated style rules.

Nimastic committed:
Merge pull request #11 from Nimastic/branch-A-CheckStyle

Add CheckStyle and fix coding style violations

realqijun commented on src/main/java/jeriel/command/DeleteCommand.java:
```diff
@@ -0,0 +1,36 @@
+package jeriel.command;
+import jeriel.task.*;
+import jeriel.util.*;
```
Might have to explicitly state which classes you're importing, same for the other files

realqijun commented on src/main/java/jeriel/command/FindCommand.java:
```diff
@@ -0,0 +1,32 @@
+
+    private String keyword;
+
+    public FindCommand(String keyword) {
```
Missing some javadoc comments for public constructors 😜

realqijun commented on src/main/java/jeriel/task/Event.java:
```diff
@@ -0,0 +1,42 @@
+package jeriel.task;
+import jeriel.command.*;
+import jeriel.util.*;import java.time.LocalDateTime;
```
Do seperate your import statements

realqijun commented on src/main/java/jeriel/task/Event.java:
```diff
@@ -0,0 +1,42 @@
+     */
+    @Override
+    public String toSaveFormat() {
+        return "[E]" + super.toString() + " (from: " + from.format(DateTimeFormatter.ofPattern("MMM d yyyy h:mm a")) + " to: " + to.format(DateTimeFormatter.ofPattern("MMM d yyyy h:mm a")) + ")";
```
Line is too long, consider doing something like this:
```
public String toSaveFormat() {
    return "[E]" + super.toString() + " (from: " 
         + from.format(DateTimeFormatter.ofPattern("MMM d yyyy h:mm a")) + " to: " 
         + to.format(DateTimeFormatter.ofPattern("MMM d yyyy h:mm a")) + ")";
```

realqijun commented on src/main/java/jeriel/util/Storage.java:
```diff
@@ -0,0 +1,84 @@
+            Task task;
+
+            switch (parts[0]) {
+                case "T":
```
If the save file is corrupted, will this still work? (Like the save file has just a "T")

realqijun commented on src/main/java/jeriel/util/Parser.java:
```diff
@@ -0,0 +1,40 @@
+        String command = commandAndArgs[0];
+        String arguments = commandAndArgs.length > 1 ? commandAndArgs[1] : "";
+
+        switch (command) {
+            case "list":
+                return new ListCommand();
+            case "mark":
+                return new MarkCommand(arguments);
+            case "unmark":
+                return new UnmarkCommand(arguments);
+            case "todo":
+                return new AddTodoCommand(arguments);
+            case "deadline":
+                return new AddDeadlineCommand(arguments);
+            case "event":
+                return new AddEventCommand(arguments);
+            case "find":
+                return new FindCommand(arguments);
+            case "delete":
+                return new DeleteCommand(arguments);
+            case "bye":
+                return new ExitCommand();
+            default:
+                throw new JerielException("I'm sorry, but I don't know what that means :-(");
```
Nice use of switch cases here

ritulkrsingh replied:
For this course we aren't supposed to indent `case` clauses :)

realqijun reviewed (commented):
Looks good overall.
Changes to coding standard issues requested. 🤩🤩🤩

ritulkrsingh commented on src/main/java/jeriel/util/Ui.java:
```diff
@@ -0,0 +1,116 @@
+            System.out.println(" " + (i + 1) + ". " + tasks.get(i));
+        }
+        System.out.println("____________________________________________________________");
+    }
```
Maybe this can be implemented in TaskList class as toString?

ritulkrsingh reviewed (commented):
LGTM, minor nitpicks here and there...

Nimastic committed:
Update README.md

Nimastic committed:
Update README.md

Nimastic committed:
Fix GUI Crashing

Nimastic committed:
Merge branch 'master' of https://github.com/Nimastic/ip

Nimastic committed:
Update README.md

Nimastic committed:
Add Help Command

Nimastic committed:
Merge pull request #13 from Nimastic/branch-level-10

Add Help Command

Nimastic committed:
Fixes

Nimastic committed:
Add New Jar

Nimastic committed:
Add assertions in Parser class to validate command inputs

- Added assertions to ensure that the 'fullCommand' is neither null nor empty.
- Ensured that the command part of the input is non-empty after splitting and trimming.
- This ensures that the program catches invalid or unexpected inputs early during the parsing process.
- Refactored the command parsing logic to improve code clarity and robustness.

Nimastic committed:
Merge pull request #14 from Nimastic/branch-A-Assertions

Add assertions in Parser class to validate command inputs

Nimastic committed:
Refactor DialogBox class to improve code quality

- Introduced constants for image size and circular clip dimensions to replace magic numbers.
- Extracted image view creation logic into a helper method to avoid duplication.
- Improved variable names for better clarity (renamed 'db' to 'dialogBox').
- Encapsulated 'text' and 'displayPicture' fields as final to ensure immutability.
- Extracted the flip logic into a separate method to adhere to the DRY principle.
- Simplified alignment handling for user and bot dialogs, ensuring consistent code structure.

Nimastic committed:
Merge pull request #15 from Nimastic/branch-A-CodeQuality

Refactor DialogBox class to improve code quality

Nimastic committed:
Refactor TaskList class to use Java Streams

- Refactored methods in TaskList class to incorporate Java Streams for improved readability and performance.
- Added a filterTasks method to allow filtering tasks based on a condition using Streams.
- Implemented tasksToString method to convert the task list to a formatted string using Streams.
- Retained compatibility with existing code by keeping tasks as an ArrayList.
- This update improves code quality and prepares the codebase for more functional operations using Streams.

Nimastic committed:
Merge pull request #16 from Nimastic/branch-A-Streams

Refactor TaskList class to use Java Streams

Nimastic committed:
Add CI workflow with GitHub Actions

Nimastic committed:
Add 'find' command to HelpCommand

- Updated HelpCommand.java to include the 'find' command in the list of available commands.
- The 'find <keyword>' command helps users search for tasks that contain the specified keyword.

Nimastic committed:
Improve GUI with better alignment and padding

- Messages are now aligned asymmetrically: user messages are right-aligned, bot messages left-aligned
- Added error highlighting for invalid commands in red
- Profile pictures are circular and use less space
- Enabled window resizing and adjusted padding

Nimastic committed:
Add Personality

Nimastic committed:
Update FindCommand.java

Nimastic committed:
Add additional JUnit tests for TaskList and FindCommand

- Updated `FindCommandTest` to handle cases with multiple tasks, no matching tasks, and an empty task list
- Replaced abstract `Task` instantiation in tests with concrete classes (`Todo`)
- Added more comprehensive tests in `TaskListTest` to verify task addition, deletion, and retrieval
- Ensured that the test cases cover edge cases, such as empty task lists and case-insensitive search

Nimastic committed:
Add Continuous Integration (CI) Support and Cross-Platform JavaFX Setup Using AI

- Added GitHub Actions CI workflow (`ci.yml`) to automate builds and tests.
  - The workflow runs on different OS environments: Windows, macOS, and Linux.
  - Ensures consistent testing across platforms using JavaFX.
  - Included job steps for setting up JavaFX SDK and Gradle for each OS.

- Updated `build.gradle` to support cross-platform JavaFX compilation.
  - Added detection logic for OS to set the correct JavaFX SDK path (`javaFxSdkPath`) based on the system’s operating system (Windows, macOS, Linux).
  - Configured Gradle to include JavaFX modules during both compile and runtime phases.
  - Ensured compatibility by adjusting `jvmArgs` and `compilerArgs` to include JavaFX modules dynamically based on the OS.
  - Enhanced `shadowJar` configuration to package the JAR with the necessary JavaFX runtime dependencies.

Nimastic committed:
Create AI.md

Nimastic committed:
Add User Guide and UI

Nimastic committed:
Update Readme

Nimastic committed:
Update README.md

Nimastic committed:
Update README.md

Nimastic committed:
Update README.md

Nimastic committed:
Update README.md

Nimastic committed:
Update README.md

Nimastic committed:
Update README.md

Nimastic committed:
Update README.md

Nimastic committed:
Update README.md

Nimastic committed:
Update README.md

Nimastic committed:
Update README.md

Nimastic committed:
Update README.md

Nimastic committed:
Update README.md

Nimastic committed:
Update README.md

Nimastic committed:
Update README.md

Nimastic committed:
Update README.md

Nimastic committed:
Update README.md

Nimastic committed:
Update README.md
